modoboa_amavis/lib.py

```diff
@@ -395,3 +397,80 @@ def setup_manual_learning_for_mbox(mbox):
                 create_user_and_use_policy(alias, policy)
             result = True
     return result
+
+
+def split_address(address):
+    assert isinstance(address, six.text_type),\
+        "address should be of type %s" % six.text_type.__name__
+    if "@" not in address:
+        local_part = address
+        domain = ""
+    else:
+        local_part, domain = address.split("@", 1)
+    return (local_part, domain)
+
+
+def split_local_part(local_part, delimiter=None):
+    assert isinstance(local_part, six.text_type),\
+        "local_part should be of type %s" % six.text_type.__name__
+    assert isinstance(delimiter, six.text_type) or delimiter is None,\
+        "delimiter should be of type %s" % six.text_type.__name__
+    if delimiter is None:
+        conf = dict(param_tools.get_global_parameters("modoboa_amavis"))
+        delimiter = conf["recipient_delimiter"]
+    extension = ""
+    if local_part.lower() in ["postmaster", "mailer-daemon", "double-bounce"]:
+        # never split these special addresses
+        pass
+    elif (
+        delimiter == "-"
+        and (local_part.startswith("owner-")
+             or local_part.endswith("-request"))
+    ):
+        # don't split owner-* or *-request if - is the delimiter, they are
+        # special addresses used by mail lists.
+        pass
+    elif (
+        delimiter
+        and delimiter in local_part
+        and local_part[0] != delimiter
+        and local_part[-1] != delimiter
+    ):
+        local_part, extension = local_part.split(delimiter, 1)
+
+    return (local_part, extension)
```

Comment:
Same here. Maybe we could improve the function available in modoboa.lib ?

Reply:
fixed, see modoboa/modoboa#1360